src/components/App.tsx

```diff
@@ -35,13 +40,13 @@ const App: React.FC = (): JSX.Element => {
   const handleSubmitPremises = (premises: string) => {
     setPremises(premises)
     setTree(parsePremises(premises.split(',')))
+    setRow(tree.forest.length + 1)
```

Comment:
As above, the length of the forest is always 1, so this doesn't quite work.

Try it with the default example. You get two 3s.

Reply:
You could try having parsePremises return the number of premises and using that.